File: BBB.Framework.Dto/Core/EncounterPositionDto.cs
namespace BBB.Framework.Dto.Core
{
    public sealed class EncounterPositionDto
    {
        public Guid EncounterPositionKey { get; set; }

        public string Name { get; set; } = string.Empty;

        public int SortOrder { get; set; }

        public string Description { get; set; } = string.Empty;

        public Guid EncounterKey { get; set; }
    }
}

File: BBBRaiderWeb/src/models/EncounterPositionDto.ts
import { logTypeMissmatch, parseNumber, parseString } from '../helpers/model'

export class EncounterPositionDto {
  encounterPositionKey: string = ''
  name: string = ''
  sortOrder: number = 0
  description: string = ''
  encounterKey: string = ''

  static fromJSON(obj: any): EncounterPositionDto {
    if (!obj) {
      logTypeMissmatch('EncounterPositionDto', obj)
      return new EncounterPositionDto()
    }

    return {
      encounterPositionKey: parseString(obj.encounterPositionKey),
      name: parseString(obj.name),
      sortOrder: parseNumber(obj.sortOrder),
      description: parseString(obj.description),
      encounterKey: parseString(obj.enconterKey),
    }
  }
}
